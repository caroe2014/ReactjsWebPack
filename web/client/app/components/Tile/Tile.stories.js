// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

/* eslint-disable max-len */
import React from 'react';
import { storiesOf } from '@storybook/react';
import Tile from '.';

storiesOf('Tile Component', module)
  .add('With Images', () => {
    return (
      <Tile image='http://eatstreet.co.id/wp-content/uploads/2016/11/cropped-MASTER-FINAL_LOGO-OVAL-EAT-STREET-1.png'
        title='Eat Street' />
    );
  })
  .add('Without Images', () => {
    return (
      <Tile title='Eat Street' />
    );
  })
  .add('With Images and children', () => {
    return (
      <Tile image='http://eatstreet.co.id/wp-content/uploads/2016/11/cropped-MASTER-FINAL_LOGO-OVAL-EAT-STREET-1.png'
        title='Eat Street'>
        <div>test</div>
      </Tile>
    );
  })
  .add('Without Images and children', () => {
    return (
      <Tile title='Eat Street'>
        <div>test</div>
      </Tile>
    );
  })
  .add('With handleOnClick', () => {
    return (
      <Tile image='http://eatstreet.co.id/wp-content/uploads/2016/11/cropped-MASTER-FINAL_LOGO-OVAL-EAT-STREET-1.png'
        title='Eat Street' OnClick/>
    );
  });
