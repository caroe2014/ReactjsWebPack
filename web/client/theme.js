// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
