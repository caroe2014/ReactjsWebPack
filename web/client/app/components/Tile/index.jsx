// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component } from 'react';
import styled, { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme';
import PropTypes from 'prop-types';
import { Flex, Text, Box } from 'rebass';
import { Trans } from 'react-i18next';
import DynamicImage from 'web/client/app/components/DynamicImage';
const Container = styled(Flex)`
  flex-wrap: no-wrap;
  flex-direction: column;
  flex: 1 1 auto;
  box-shadow: ${props => {
    if (!props['data-border']) return 'none';
    return props.theme.shadow;
  }};
  ${props => props.theme.mediaBreakpoints.desktop`
  min-height: ${props => {
    if (props.iscart) return '110px';
    return '170px';
  }};
  margin-bottom: ${props => {
    if (props.iscart) return '0px';
    return '20px';
  }};
  `};
  position: relative;
  width: 100%;
  max-width: 570px;
  cursor: ${props => !props.disabled && props['data-action'] ? 'pointer' : 'initial'};
  
  background-color: white;
  ${props => props.theme.mediaBreakpoints.tablet`margin-bottom: 10px; min-height: 170px;`};
  ${props => props.theme.mediaBreakpoints.mobile`margin-bottom: 10px; min-height: 110px;`};
  .custom-time{
    font-weight: 400;
  }
  &:hover {
    .closed-overlay{
      opacity: 1;
      visibility: visible;
    }
    .closes-text{
      font-weight: 300 !important;
    }
    .custom-title{
      font-weight: 500 !important;
    }
    .custom-time{
      font-weight: bold;
    }
    ${props => props.theme.mediaBreakpoints.desktop`
      ${props => {
    return `box-shadow: 2px 3px 10px 0px rgba(0, 0, 0, .30);
          -webkit-box-shadow: 2px 3px 10px 0px rgba(0, 0, 0, .30);
          -moz-box-shadow: 2px 3px 10px 0px rgba(0, 0, 0, .30);
          .title-hover{
            color: ${props.theme.colors.textHoverColor} !important;`;
  }}
`};
  };
`;
const TopContainer = styled(Flex)`
  flex: 1 1 auto;
  background-color: #FFF;
  @media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) {
    min-height: 170px;
  }
`;
const ImageContainer = styled(DynamicImage)`
  border: 0.5px solid lightgray;
`;
ImageContainer.displayName = 'ImageContainer';
const DetailContainer = styled(Flex)`
  flex: 1 1 auto;
  height:auto;
  padding: ${props => {
    if (props.iscart) return '10px 0px';
    return '10px';
  }};
  
  word-wrap: break-word;
`;
const BoxContainer = styled(Flex)` 
  height: 170px;
  width: 170px;
  position: relative;
${props => props.theme.mediaBreakpoints.mobile`
    width: ${props => {
    if (props.imageresponsive) return '120px';
    return '190px';
  }};
    height: ${props => {
    if (props.imageresponsive) return '120px';
    return '190px';
  }};
`};
`;
const TextContainer = styled(Text)`
  color: ${props => props.theme.colors.primaryTextColor};
  font-size: 20px;
  font-weight: bold;
  ${props => props.theme.mediaBreakpoints.tablet`font-size: 18px;`};
  ${props => props.theme.mediaBreakpoints.mobile`font-size: 18px;`};
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;
const Overlay = styled(Box)`
  width: 100%;
  height: 100%;
  position: absolute;
  opacity: 0.5;
  z-index: 3;
  background-color: ${props => props.theme.colors.light};
`;
const ClosedOverlay = styled(Box)`
  width: 100%;
  height: 100%;
  position: absolute;
  display: block;
  z-index: 4;
  background: rgba(0, 0, 0, 0);
  left: 0;
  top: 0;
  padding: inherit;
  & > div {
    display: flex;
    opacity: 0;
    visibility: hidden;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    background: rgba(0, 0, 0, 0.55);
    height: 100%;
    width: 100%;
    color: ${props => props.theme.colors.reverseTextColor};
  }
${props => props.theme.mediaBreakpoints.tablet`
    display: ${props => !props.disabled ? 'none' : 'block'};
`}
`;

class Tile extends Component {
  constructor (props) {
    super(props);
    this.handleOnClick = this.handleOnClick.bind(this);
    this.handleKeyDown = this.handleKeyDown.bind(this);
  }
  static defaultProps = {
    isEnabled: true,
    hasBorder: true
  }
  handleOnClick (e) {
    if (this.props.handleClick && this.props.isEnabled) {
      this.props.handleClick(this.props.id);
    }
  }
  handleKeyDown (e) {
    // TODO: make this usable by all clickable elements,
    //  pass in caller's onclickhandler (13 is keycode for enter key, 32 is spacebar)
    if (e.which === 13 || e.which === 32) {
      this.handleOnClick();
    }
  }
  render () {
    const {
      classContext,
      image,
      title,
      titleStyle,
      children,
      footer,
      handleClick,
      isEnabled,
      hasBorder,
      closedoverlay,
      isHover,
      ariaLabel,
      tabIndex,
      ...rest } = this.props;
    return (
      <ThemeProvider className='tile-container' theme={theme}>
        <Container
          {...rest}
          className={classContext}
          flexWrap='nowrap'
          role='button'
          onClick={this.handleOnClick}
          onKeyDown={this.handleKeyDown}
          data-action={handleClick !== undefined}
          disabled={!isEnabled}
          data-border={hasBorder}
          is-hover={isHover}
        >
          <TopContainer {...rest} className='top-container' width={1} id={`top-container${classContext}`}>
            { !isEnabled &&
              <Overlay className='overlay' />
            }
            {image &&
              <BoxContainer {...rest} className='box-container' width={[0.4]} id={`box-container${rest.id}`} >
                <ImageContainer
                  className='image'
                  src={image}
                  bgSize='100% 100%'
                  alt=''
                />
                { closedoverlay &&
                <ClosedOverlay disabled={!isEnabled}>
                  <div className='closed-overlay'>
                    {isEnabled ? <Trans i18nKey='TILE_OPEN'/> : <Trans i18nKey='TILE_CLOSED'/>}
                  </div>
                </ClosedOverlay>
                }
              </BoxContainer>
            }
            <DetailContainer
              {...rest}
              className='detail-container'
              flexDirection='column'
              alignItems={children ? 'flex-start' : 'center'}
              justifyContent={children ? 'flex-start' : 'center'}
              id={`detail-container${classContext}`}
              width={image ? [0.6] : [1]}
            >
              <TextContainer className='title-hover tile-tile'
                aria-label={ariaLabel} tabIndex={tabIndex}
                style={titleStyle} handlehover={handleClick && 'true'}>
                {title}
              </TextContainer>
              {children && children}
            </DetailContainer>
          </TopContainer>
          {footer && footer}
        </Container>
      </ThemeProvider>
    );
  }
}
Tile.propTypes = {
  handleClick: PropTypes.func,
  image: PropTypes.string,
  title: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]),
  titleStyle: PropTypes.object,
  footer: PropTypes.element,
  isEnabled: PropTypes.bool,
  hasBorder: PropTypes.bool
};
export default Tile;
