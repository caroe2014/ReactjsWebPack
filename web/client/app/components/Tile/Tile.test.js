// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme.js';
import Tile from '.';

configure({ adapter: new Adapter() });
/* global describe, it, expect, beforeEach */
/* eslint-disable max-len */
describe('Tile Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Tile image='http://eatstreet.co.id/wp-content/uploads/2016/11/cropped-MASTER-FINAL_LOGO-OVAL-EAT-STREET-1.png'
      title='Eat Street' />
    );
  });

  it('does a snapshot test', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('when image props received', () => {
    expect(wrapper.find('ImageContainer')).toHaveLength(1);
  });
  it('when image props not received', () => {
    wrapper = shallow(<Tile title='Eat Street' />);
    expect(wrapper.find('ImageContainer')).toHaveLength(0);
  });
  it('when childrens given render without error', () => {
    wrapper = shallow(<Tile title='Eat Street'><div>test</div></Tile>);
    expect(wrapper.find('ImageContainer')).toHaveLength(0);
  });
});
