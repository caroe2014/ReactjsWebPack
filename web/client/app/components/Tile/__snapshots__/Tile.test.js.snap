// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile Component does a snapshot test 1`] = `
<ThemeProvider
  className="tile-container"
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <Styled(Flex)
    data-action={false}
    data-border={true}
    disabled={false}
    flexWrap="nowrap"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    <Styled(Flex)
      className="top-container"
      id="top-containerundefined"
      width={1}
    >
      <Styled(Flex)
        className="box-container"
        id="box-containerundefined"
        width={
          Array [
            0.4,
          ]
        }
      >
        <ImageContainer
          alt=""
          bgSize="100% 100%"
          className="image"
          src="http://eatstreet.co.id/wp-content/uploads/2016/11/cropped-MASTER-FINAL_LOGO-OVAL-EAT-STREET-1.png"
        />
      </Styled(Flex)>
      <Styled(Flex)
        alignItems="center"
        className="detail-container"
        flexDirection="column"
        id="detail-containerundefined"
        justifyContent="center"
        width={
          Array [
            0.6,
          ]
        }
      >
        <Styled(Text)
          className="title-hover tile-tile"
        >
          Eat Street
        </Styled(Text)>
      </Styled(Flex)>
    </Styled(Flex)>
  </Styled(Flex)>
</ThemeProvider>
`;
